unattended-upgrade

```diff
@@ -431,15 +445,27 @@ def should_stop():
     if SIGNAL_STOP_REQUEST:
         logging.warning("SIGNAL received, stopping")
         return True
-    if apt_pkg.config.find_b("Unattended-Upgrade::OnlyOnACPower", True) and \
-       subprocess.call("on_ac_power") == 1:
-        logging.warning("System is on battery power, stopping")
-        return True
+    try:
+        if apt_pkg.config.find_b("Unattended-Upgrade::OnlyOnACPower", True) \
+           and subprocess.call("on_ac_power") == 1:
+            logging.warning("System is on battery power, stopping")
+            return True
+    except FileNotFoundError:
+        log_once(
+            _("Checking if system is running on battery is skipped. Please "
+              "install powermgmt-base package to check power status and skip "
+              "installing updates when the system is running on battery."))
     if apt_pkg.config.find_b(
             "Unattended-Upgrade::Skip-Updates-On-Metered-Connections", True):
-        if NetworkMonitor.get_network_metered(NetworkMonitor.get_default()):
-            logging.warning(_("System is on metered connection, stopping"))
-            return True
+        try:
+            if NetworkMonitor.get_network_metered(
```

Comment:
Same here as above, a helper like "is_metered()" and then dynamically patching it. Feels slightly nicer to me because then reading the checks is less distracting (because now they include the error condition) - hope this explanation makes sense(?). But might be personal preferences and certainly not a blocker.